src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -315,313 +346,324 @@ void func_80B67540(EnSth* this, PlayState* play) {
     SkelAnime_Update(&this->skelAnime);
 
     if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
-        func_80B67984(this, play);
-        this->actionFunc = func_80B67C1C;
-    } else if (func_80B6703C(this, play)) {
+        EnSth_GetInitialSwampSpiderhouseText(this, play);
+        this->actionFunc = EnSth_HandleSwampSpiderhouseConversation;
+    } else if (EnSth_DetectSpeakingRange(this, play)) {
         this->actor.textId = 0;
         func_800B8614(&this->actor, play, 110.0f);
     }
 }
 
-void func_80B67E20(Actor* thisx, PlayState* play) {
+void EnSth_OceanSpiderhouseWaitForTokens(Actor* thisx, PlayState* play) {
     EnSth* this = THIS;
 
-    if (Inventory_GetSkullTokenCount(play->sceneId) >= 30) {
-        this->actor.update = func_80B680A8;
-        this->actor.draw = func_80B6849C;
+    if (Inventory_GetSkullTokenCount(play->sceneId) >= STH_OCEAN_SPIDER_TOKENS_REQUIRED) {
+        this->actor.update = EnSth_Update;
+        this->actor.draw = EnSth_Draw;
         this->actor.flags |= ACTOR_FLAG_1;
     }
 }
 
-void EnSth_Update(Actor* thisx, PlayState* play) {
+/**
+ * Dual object actors have to wait for the object to finish loading,
+ * this dev chose to use a temporary update instead of temporary acitonFunc
+ */
+void EnSth_WaitForObject(Actor* thisx, PlayState* play) {
     s32 pad;
     EnSth* this = THIS;
 
-    if (Object_IsLoaded(&play->objectCtx, this->unk_29E)) {
-        this->actor.objBankIndex = this->unk_29E;
+    if (Object_IsLoaded(&play->objectCtx, this->mainObjectId)) {
+        this->actor.objBankIndex = this->mainObjectId;
         Actor_SetObjectDependency(play, &this->actor);
 
-        if (ENSTH_GET_100(&this->actor)) {
-            SkelAnime_InitFlex(play, &this->skelAnime, &object_ahg_Skel_005998, &ovl_En_Sth_Anim_0045B4,
-                               this->jointTable, this->morphTable, 16);
-            Animation_PlayLoop(&this->skelAnime, &ovl_En_Sth_Anim_0045B4);
-            this->unk_29A = 1;
+        if (STH_GET_SWAMP_BODY(&this->actor)) {
+            // as this version is unused, we don't know what scene they were meant for
+            // except: WEEKEVENTREG_34_40 is used in swamp spiderhouse
+            SkelAnime_InitFlex(play, &this->skelAnime, &gAhgSkel, &gEnSthBendingDownAnim, this->jointTable,
+                               this->morphTable, AHG_LIMB_MAX);
+            Animation_PlayLoop(&this->skelAnime, &gEnSthBendingDownAnim);
+            this->curAnimIndex = STH_ANIM_BENDING_DOWN;
             if (CHECK_WEEKEVENTREG(WEEKEVENTREG_34_10) || CHECK_WEEKEVENTREG(WEEKEVENTREG_34_20) ||
-                CHECK_WEEKEVENTREG(WEEKEVENTREG_34_40) || (Inventory_GetSkullTokenCount(play->sceneId) >= 30)) {
-                func_80B670A4(this, 3);
+                CHECK_WEEKEVENTREG(WEEKEVENTREG_34_40) ||
+                (Inventory_GetSkullTokenCount(play->sceneId) >= STH_SWAMP_SPIDER_TOKENS_REQUIRED)) {
+                EnSth_SwitchAnimation(this, STH_ANIM_WAITING);
```

Comment:
I'm a bit confused; this *is* used, correct? I think what's actually happening here is that this actor gets instantly killed if the skull count is below 30, and if it's above 30, his animation changes. Thus, we never see his bending down animation. Do I have that right? If so, then I think the comment should be updated to describe this, since it's kind of confusing as-is.

Reply:
Oh old comment, not accurate yeah I need to fix this.